rails/lib/tasks/lara2.rake

```diff
@@ -103,4 +103,71 @@ namespace :lara2 do
     end
   end
 
+  desc "migrates external report urls"
+  task :migrate_external_report_urls do
+    # "Class Dashboard" is the existing LARA class dashboard an should be changed to "Activity Player Dashboard"
+
+    begin
+      ap_tool = Tool.find_by_name("ActivityPlayer")
+      raise "No ActivityPlayer tool found" if ap_tool.nil?
+
+      ap_report = ExternalReport.find_by_name("Activity Player Report")
+      raise "No 'Activity Player Report' external report found" if ap_report.nil?
+
+      lara_dashboard_report = ExternalReport.find_by_name("Class Dashboard")
+      raise "No 'Class Dashboard' external report found" if lara_dashboard_report.nil?
+
+      ap_dashboard_report = ExternalReport.find_by_name("Activity Player Dashboard")
+      raise "No 'Activity Player Dashboard' external report found" if ap_dashboard_report.nil?
+
+      puts "Are you sure you want to PERMANENTLY migrate the external activity reports for AP activities? Enter 'YES' to confirm:"
+      if STDIN.gets.chomp == "YES"
+        puts "Migrating reports"
+
+        # first set the default report type for AP activities to be the "Activity Player Report"
+        ap_report.default_report_for_source_type = ap_tool.source_type
+        ap_report.save!
```

Comment:
I'm not sure if this will work. It would be nice, but I think there was some problem with this, which is why we haven't done it before.  Did you test it on some local or remote Portal+LARA?

Reply:
You are right - I read the code and it looked like that would set the "Report" button but I just checked an activity set to AP and the report button doesn't show.  I'll update the task.

Reply:
@scytacki Fixed in https://github.com/concord-consortium/rigse/pull/1131/commits/b2953190583edcd681815c0b7edbfe5d22dbba1d